pkg/kv/kvserver/replica_consistency.go

```diff
@@ -500,9 +500,27 @@ func CalcReplicaDigest(
 	var timestampBuf []byte
 	hasher := sha512.New()
 
+	// Request quota from the limiter in chunks of quotaBatchSize, to amortize the
+	// overhead of the limiter when reading many small KVs.
+	var budget int64
+	const quotaBatchSize = int64(4 << 10) // 4 KiB
```

Comment:
Maybe something like 64 KiB would be better, still less than 1% of the total limit. Given that there are only a few parallel tasks, it's ok to overshoot by 1%.